kernel/kernel-api/src/main/java/io/delta/kernel/internal/checksum/ChecksumUtils.java

```diff
@@ -0,0 +1,128 @@
+/*
+ * Copyright (2025) The Delta Lake Project Authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.delta.kernel.internal.checksum;
+
+import static io.delta.kernel.internal.actions.SingleAction.CHECKPOINT_SCHEMA;
+import static io.delta.kernel.internal.util.Preconditions.checkState;
+
+import io.delta.kernel.engine.Engine;
+import io.delta.kernel.exceptions.ChecksumAlreadyExistsException;
+import io.delta.kernel.internal.SnapshotImpl;
+import io.delta.kernel.internal.actions.AddFile;
+import io.delta.kernel.internal.actions.DomainMetadata;
+import io.delta.kernel.internal.replay.CreateCheckpointIterator;
+import io.delta.kernel.internal.stats.FileSizeHistogram;
+import java.io.IOException;
+import java.nio.file.FileAlreadyExistsException;
+import java.time.Instant;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Optional;
+import java.util.concurrent.atomic.LongAdder;
+import java.util.stream.Collectors;
+
+public class ChecksumUtils {
+  /**
+   * Computes the state of a Delta table and writes a checksum file based on the provided snapshot.
+   *
+   * <p>This method iterates through the table's data using a CreateCheckpointIterator to calculate:
```

Comment:
nit: maybe don't mention the _exact_ iteration mechanism (i.e. CreateCheckpointIterator) -- we may rename that or change it later? i.e. keep this comment high-level and readers can read the code for more detail?